src/main/java/bootcamp/services/twitter4j/TwitterResourceService.java

```diff
@@ -62,6 +76,7 @@ public Tweet statusToTweet(Status status) {
      */
     public Optional<Tweet> postTweet(Message post) {
         logger.info("TwitterResourceService called postTweet");
+        cache.invalidateAll(); // clear cache so that the new message is there when it's loaded again
```

Comment:
this isn't any different from before

Reply:
In this version, the cache will be cleared regardless of the success or validity of the post tweet, instead of clearing it only after the call had succeeded. This way, even if post is invalid, and even if there's a different issue with Twitter, the cache will be invalidated, whereas earlier, if an error was thrown by the method's core functionality, it would not have reached the cache.invalidateAll(). I may have misunderstood what you wanted me to change. Did you have something else in mind?

Reply:
Sorry, I must have misunderstood what you meant earlier. I thought you meant that you wanted me to clear it regardless of the validity of post. Now the cache clearing is after the both getting the post content and calling twitter, which would mean it will not be cleared unless everything succeeds.